보틀월드 팔로잉 팔로워 목록 네트워크 연결

xwoud opened the pull request:
## 🤗 What is this PR?
- 보틀월드 팔로잉 팔로워 목록 네트워크 연결
- issue: #60 

## ✅ Test Checklist

- [ ] 팔로잉 api 연결
- [ ] 팔로우 api 연결
- [ ] empty view 

## 💬 Comment
- 일단 로딩 넣고 싶긴 했는데 .. 물어보고 넣어야할 것 같아서 아직 안넣었어요
- 그리고 서버 연결 후에 팔로우 팔로잉 수를 swipebar에 표시해야해서 delegate 2개를 연결해서 전달하는데 더 좋은 방법이 있다면 알려주숍 ;ㅅ;

xwoud committed:
[✨ feature] 보틀월드 팔로워 API Target 추가

feat: #60

xwoud committed:
[✨ feature] 보틀월드 팔로워 API Target 추가

feat: #60

xwoud committed:
[✨ feature] 보틀월드 팔로잉 API Target 추가

feat: #60

xwoud committed:
[✨ feature] 보틀월드 팔로잉 API Target 추가

feat: #60

xwoud committed:
[✨ feature] 보틀월드 팔로워 API 메소드 추가

feat: #60

xwoud committed:
[:bulb: comment] api 분리를 위한 주석 추가

feat: #60

xwoud committed:
[✨ feature] 보틀월드 팔로잉 API 메소드 추가

feat: #60

xwoud committed:
[:sparkles: feature] 팔로워 api 연결 후 데이터 적용

body: 각 탭별로 값이 없을 때 다른 empty view가 뜨게 설정

feat: #60

xwoud committed:
[:sparkles: feature] 팔로워 api 연결 후 데이터 적용

body: 각 탭별로 값이 없을 때 다른 empty view가 뜨게 설정

feat: #60

xwoud committed:
[:sparkles: feature] 팔로잉 api 연결 후 데이터 적용

feat: #60

xwoud committed:
[:sparkles: feature] 팔로잉 api 연결 후 데이터 적용

feat: #60

xwoud committed:
[:fire: remove] 사용하지 않는 코드 삭제

feat: #60

xwoud committed:
[:art: style] enum 클래스 밖으로 꺼내기

feat: #60

xwoud committed:
[:fire: remove] 더미데이터에 쓴 함수 삭제

feat: #60

xwoud committed:
[:art: style] 중복되는 코드 함수로 제작

feat: #60

xwoud committed:
[:sparkles: feature] 팔로워 팔로잉 수 swipe bar에 전달 기능 추가

body: 전달 시 클릭이 해제되는 문제 있어서 재클릭을 위해 selectItem 을 사용

feat: #60

xwoud committed:
[:art: style] 프로퍼티 명 변경

feat: #60

xwoud committed:
[:lipstick: layout] 팔로잉 팔로우 바껴있는거 수정

feat: #60

Juhyeoklee commented:
으으으으으으으음 UserListView에서 데이터 통신이 이루어 지다보니 구조가 마니 복잡해 지고 있다는 느낌이 마니 들었어요.
차라리 UserLisrView가 아니라 BottleWorldListCell에서 데이터를 받아서 UserListView에 값을 뿌려주고 카운트 값을 BottleWorldVC에 delegate나 closure로 넘겨주는게 delegate depth를 줄이고 구조가 조금더 간편해 질것 같은 느낌이 들어요!!

혹시 UserListView에서 데이터 를 받아ㅏ오는 이유가 있을까요 ?!

xwoud committed:
[:zap: improve] 데이터 통신 구조를 cell로 이동

body: 기존의 UserListView에서 통신 후 데이터를 뿌려주는 형식이 복잡해져서 Cell로 데이터 통신을 이동

feat: #60

xwoud committed:
[:zap: improve] 데이터 통신 구조를 cell로 이동

body: 기존의 UserListView에서 통신 후 데이터를 뿌려주는 형식이 복잡해져서 Cell로 데이터 통신을 이동

feat: #60

xwoud committed:
[:zap: improve] 데이터 통신 구조를 cell로 이동

body: 기존의 UserListView에서 통신 후 데이터를 뿌려주는 형식이 복잡해져서 Cell로 데이터 통신을 이동

feat: #60

xwoud committed:
[:lipstick: layout] textField 문구 수정
feat: #60

xwoud committed:
이메일 관련 화면 수정 및 삭제 (#65)

이메일 관련 화면 수정 및 삭제

xwoud committed:
설정 화면 스크립션 수정 (#67)

* [✨ sparkles] 설정화면에서 유저 정보 받아오는 network case 추가

feat: #66

* [:sparkles: feature] 추가한 UserInfo 받아오는 네트워크 구조체를 이용하여 Static Class 생성

feat: #66

* [:zap: improve] UIBarButtonItem 의 touch action이 먹히지 않아 lazy 상태로 변경

feat: #66

* [:sparkles: feature] 홈 화면에서 햄버거 바 클릭 시 설정 화면으로 이동하는 기능 추가

feat: #66

* [:sparkles: feature] 닉네임 클릭 시 계정 설정 화면으로 이동
body: 이전에는 버튼을 제작해 닉네임만 클릭 시 이동하였지만, 클릭 범위가 늘어남에 따라 UIView자체에 tap gesture를 주는 형식으로 변경

feat: #66

* [:lipstick: layout] button에서 Label로 변경

* [:lipstick: layout] 닉네임 1글자를 보여주는 로직 수정
body: 맨 앞 글자가 _인 경우에는 두번째 글자를 보여주는 로직추가

feat: #66

* [:lipstick: layout] storyboard 상의 layout 변경

* [:sparkles: feature] 설정에서 각 탭 클릭 시 외부 사이트로 연결 기능 추가
body: 인스타그램 제외 링크 변경 필요

feat: #66

* [:art: style] 닉네임을 홈 화면에서 전달 받은 후 적용

feat: #66

* [:sparkles: feature] 홈 화면에서 설정 화면 개인 정보 불러오는 네트워크 연결
body: 설정 화면이 아니라 홈 화면에서 네트워크를 불러온 이유는 홈 화면에서 사용자의 닉네임을 title로 사용해야하기 때문
또한 설정 화면에서 닉네임 변경 후 dismiss로 돌아왔을 때 다시 불러오기 위해 viewWillAppear에 코드 작성

feat: #66

* [:art: style] Lint 에 맞춰 에러 수정

* [:memo: docs] 외부 링크 수정

* [:bug: bug] 500 error status code를 잡아내는 코드 수정

* [:bulb: comment] 린트 주석 추가

* [:art: style] URL 웹 사이트로 연결하는 Extension 적용

feat: #66

xwoud committed:
닉네임 설정 api 연동 (#68)

* [:sparkles: feature] 닉네임 변경 API Service 제작

feat: #51

* [:art: style] 400 에러 오는 조건문 수정

body: 닉네임에서 400 에러는 중복을 뜻함

feat: #51

* [:sparkles: feature] 가입하기 버튼 클릭 시 로그인 할 수 있는 화면으로 이동

feat: #51

* [:sparkles: feature] 미가입자 뷰에도 closure 전달

body: navigation 작동이 필요하기 때문에 전달

feat: #51

* [:zap: improve] 뷰 이동을 Navigation controller 의 이동으로 수정

feat: #51

* [:sparkles: feature] navigation extension 추가

body: 여러번 pop 할 수 있는 기능 추가

feat: #51

* [:sparkles: feature] 닉네임 변경 api 연결 후 결과에 맞춰 기능 추가

feat: #51

* [:art: style] 서버 오류 분기 처리

* [:fire: remove] 사용하지 않는 테스트 코드 삭제

* [:sparkles: feature] 닉네임 변경 시 서버 연결 후 중복 검사 및 닉네임 수정 기능 추가

* [:fire: remove] email 정보 삭제

feat: #51

* [:sparkles: feature] 기존 닉네임과 같을 시 서버 전송하지 않고 그대로 종료되게 수정

xwoud committed:
설정과 계정 설정 기능 연결  (#70)

* [:sparkles: feature] 설정 화면에서 닉네임 및 계정 공개 정보 데이터 전달

feat: #69

* [:sparkles: feature] 닉네임과 계정 공개 여부 변경 시 이전 View에 값 전달 후 적용

feat: #69

* [:sparkles: feature] 계정 설정 화면에서 설정 화면에서 돌아올 때 바뀐 유저 정보가 있다면 적용

feat: #69

xwoud committed:
계정 공개여부 toggle  (#71)

* [:sparkles: feature] 유저 계정 공개 여부 api case 생성

feat: #52

* [:sparkles: feature] 계정 공개 비공개 수정 서버 전달 기능 추가

feat: #52

* [:bulb: comment] 토큰 주석 추가

* [:sparkles: feature] 토글 버튼 클릭 시 서버 연결의 결과에 따라 토글 버튼 수정

feat: #52

* [:art: style] merge comflict 수정
dquote>
dquote> feat: #52
dquote>

Juhyeoklee committed:
알림센터 (#73)

* [:lipstick: layout] 알람센터 셀 생성

feaT: 알림센터 알림유/무 버전 #63

* [:lipstick: layout]  메인 화면 alarm center 진입점 생성

feat: 알림센터 알림유/무 버전 #63

* [:lipstick: layout] 알람센터 셀 생성

feaT: 알림센터 알림유/무 버전 #63

* [:lipstick: layout]  메인 화면 alarm center 진입점 생성

feat: 알림센터 알림유/무 버전 #63

* [:lipstick: layout] alarm center cell 생성

feat: 알림센터 알림유/무 버전 #63

* [:lipstick: layout] 알림 리액션 화면 작업

feat: 알림센터 알림유/무 버전 #63

* [:lipstick: layout] 알람센터 empty view 설정

feat: 알림센터 알림유/무 버전 #63

* [:lipstick: layout] 시작 스토리보드 설정

feat: 알림센터 알림유/무 버전 #63

* [:fire: remove] 네비게이션 버튼 코드 삭제

feat: 알림센터 알림유/무 버전 #63

* [:art: syle] 더미데이터 정리, 알람버튼 액션 정리

feat: 알림센터 알림유/무 버전 #63

* [:lipstick:  layout] 응원, 축하 토스트 화면 구현

feat: 알림센터 알림유/무 버전 #63

* [:fire: remove] whitespace 제거

feat: 알림센터 알림유/무 버전 #63

* [:art: style] setLabel extension 사용

feat: 알림센터 알림유/무 버전 #63

* [:lipstick: layout] 네비 바 와 여백 추가

feat: 알림센터 알림유/무 버전 #63

* [:lipstick: layout] 네비게이션 바 back 버튼 수정

feat: 알림센터 알림유/무 버전 #63

* [:lipstick: layout] 알람 이미지 뷰 설정

feat: 알림센터 알림유/무 버전 #63

xwoud committed:
카카오 로그인 연동 (#76)

* [:lipstick: layout] kakao 로고 이미지 추가

feat: #72

* [:card_file_box: database] lint 제외 문자열 추가

feat: #72

* [:sparkles: feature] 카카오 로그인 시 카카오톡 화면에서 다시 우리 어플로 돌아올 때를 위해 URLContexts 함수 추가

feat: #72

* [:sparkles: feature] Kakao 네이티브 앱 키 추가

feat: #72

* [:lipstick: layout] 카카오 로그인 버튼 생성

* [:sparkles: feature] 카카오 로그인 후 데이터 받아서 서버에 전송

feat: #72

* [:sparkles: feature] HomeVC에도 로그인 연결
body: 테스트 할 때를 위해 주석 처리 해둠

feat: #72

* [:sparkles: feature] 카카오 로그인 및 애플 로그인 기능 추가

Juhyeoklee committed:
챌린제 메인 대공사 (#77)

* [:lipstick: layout] 메인 화면 레이아웃 정리

feat: 메인화면 수정 #74

* [:lipstick: layout] EmptyView 설정

feat: 메인화면 수정 #74

* [:sparkles: feature] EmptyView delegate 연결

feat: 메인화면 수정 #74

* [:art: style] Empty view 설정부분 주석 처리

feat: 메인화면 수정 #74

* [:bug: bug] 챌린지 편집 버그 수정

feat: 메인화면 수정 #74

* [:lipstick: layout] empty view 설정

feat: 메인화면 수정 #74

* [:bug: bug] 레이아웃 버그 수정

feat: 메인화면 수정 #74

* [:lipstick: layout] 내 챌린지(미진행) 네트워크 연결

feat: 메인화면 수정 #74

* [:sparkles: feature]  메인화면 네트워크 연결

feat: 메인화면 수정 #74

* [:sparkles: feature] 챌린지 완료 토글 네트워크 연결

feat: 메인화면 수정 #74

* [:sparkles: feature] 내 불편함 변경 APITarget 설정

feat: 메인화면 수정 #74

* [:sparkles: feature] 내 불편함 업데이트 api 설정

feat: 메인화면 수정 #74

* [:art: style] 키보드 세팅 익스텐션 파일 분리

feat: 메인화면 수정 #74

* [:bug: bug] bug fix

feat :메인화면 수정 #74

* [:sparkles: feature] 데이터 불러오기 전 뷰 숨김 처리

feat: 메인화면 수정 #74

* [:sparkles: feature]  다른 유저 챌린지 보기 설정

feat: 메인화면 수정 #74

* [:sparkles:  feature] 온보딩 마지막 단계 넣기

feat: 메인화면 수정 #74

xwoud committed:
회원 탈퇴하기 api 연결 (#80)

* [:sparkles: feature] 유저 삭제(탈퇴) api 연결

feat: #78

* [:art: style] white space 제거

* [:sparkles: feature] 탈퇴 api 연결 후 alert에 연결

feat: #78

Juhyeoklee committed:
온보딩/온보딩4 카카오톡 로그인 연결 (#84)

* [:lipstick: layout] 온보딩4 카톡 로그인 디자인

feat: 온보딩/온보딩4 카카오톡 로그인 연결 #79

* [:sparkles: feature] 카카오 로그인 연결하기

feat: 온보딩/온보딩4 카카오톡 로그인 연결 #79

fix #79

xwoud committed:
앱 아이콘 및 어플리케이션 이름 적용 (#83)

* [:lipstick: layout] 앱 아이콘 추가

feat: #81

* [:lipstick: layout] 앱 display name 워시유어보틀 로 수정

feat: #81

Juhyeoklee committed:
UserDefaultService 만들기 (#85)

* [:sparkles: feature] UserDefaultManager 생성

feat: UserDefaultService 만들기 #82

* [:sparkles: feature] 로그인 토큰 연결

feat: UserDefaultService 만들기 #82

* [:lipstick: layout] 다크모드 제거

* [:bug: bug] 닉세팅 뷰 -> 챌린지 넘어가기 추가

feat" UserDefaultService 만들기 #82

fix: #82

Juhyeoklee committed:
축하/응원하기 API 연결 (#88)

* [:sparkles: feature] 응원하기 API 연결

feat: 축하/응원하기 API 연결 #87

* [:art: style] challenge view delegate 파일 분리

* [:art: style] ChallengeVC 네트워크 부분 파일 분리

fix #87

xwoud committed:
설정/계정설정/계정관리 1차 QA 수정 (#92)

* [:white_check_mark: test] json 테스트 파일 추가

feat: #89

* [:white_check_mark: test] json 파일 decode 하는 function 추가

feat: #89

* [:sparkles: feature] button의 tag를 이용해서 0일때는 수정을 아예 하지 않은 상태(혹은 수정 후 저장한 상태), 1일때는 수정중인 상태로 상태 구분

feat: #89

* [:fire: remove] 테스트 파일 및 코드 삭제

* [:art: style] 동일한 코드 함수로 묶어서 사용
body: 편집을 완료했는지 안했는지를 구분해서 버튼의 tag와 image를 변경해줌
feat: #89

xwoud committed:
[:bug: bug] 닉네임 값이 없을 때도 키보드의 return이 동작하는 bug 수정 (#93)

feat: #90

Juhyeoklee committed:
[:bug: bug] 챌린지 오픈 1단계 QA 이슈 수정 (#95)

feat: 챌린지오픈/1단계(편리함 작성) 1차 QA 수정 #91

xwoud committed:
스플래쉬(런치스크린) 적용 (#96)

* [:sparkles: feature] 스플래쉬 VC 추가 후 로티 기능

feat: #94

* [:memo: docs] 스플래쉬 폴더 추가

* [:wrench: configure] splash 파일 및 폴더 삭제

* [:lipstick: layout] Splash 관련 Storyboard 및 ViewController 생성

feat: #94

* [:lipstick: layout] LunchScreen에 splash 정적 이미지 삽입

feat: #94

* [:sparkles: feature] Lottie 화면에서 1바퀴 돌아간 후 token 의 유무를 체크하여 root View를 변경

feat: #94

* [:wrench: configure] Splash Lottie json 파일 추가

feat: #94

* [:sparkles: feature] 위치를 지정한 후 lottie run

feat: #94

* [:fire: remove] Lottie 사용 안하기로 해서 관련 코드 및 파일 삭제

feat: #94

* [:sparkles: feature] 스플래쉬 VC에서 1.5초 후에 넘어가는 기능 추가

feat: #94

xwoud committed:
캘린더 API 연결 및 레이아웃 변경 (#97)

* [:sparkles: feature] 캘린더 네트워크 연결 api 추가

* [:sparkles: feature] 캘린더 Api 호출을 통해 데이터 불러오기 기능 추가

feat: #75

* [:white_check_mark: test] 캘린더 구조체 변경을 위해 테스트 코드 작성중

feat: #75

* [:poop: poop] 서버 테스트코드 작성중

* [:white_check_mark: test] 서버 더미데이터를 대신해서 사용할 json 데이터 추가

feat: #75

* [:sparkles: feature] calendar api에서 id 값을 주지 않으면 기본적으로 현재 챌린지 하고 있는 데이터 전달 받기 위해 optional로 수정

feat: #75

* [:art: style] lint error 수정

* [:fire: remove] 더미데이터 삭제

feat: #75

* [:sparkles: feature] json 파일을 디코딩 해 데이터를 추출하는 함수 추가

feat: #75

* [:art: style] 현재 서버 더미데이터 오류로 인해 직접 만든 json 파일을 이용하기 위해 함수 호출 부분 수정

feat: #75

* [:fire: remove] 사용하지 않는 코드 삭제

* [:sparkles: feature] 다른 챌린지 클릭 시 또 다시 서버 연결을 통해 데이터 받아오기
body: 다른 챌린지를 클릭 했을 때 다시 서버 연결을 통해 특정 주제에 대한 불편함 챌린지 리스트를 받아오게 되는데 이미 데이터가 있다면 전체 데이터를 수정하지 않고 받아온 불편함 챌린지 리스트를 추가해주는 방식으로 개발

feat: #75

* [:white_check_mark: test] Test Json 파일 수정

feat: #75

* [:sparkles: feature] 서버 시간이 한국 시간으로 변경되어서 타임존을 ko로 수정한 후 대입

feat: #75

* [:sparkles: feature] 캘린더 api가 본인 캘린더/다른유저 캘린더 조회로 나눠져 있어서 enum 을 생성

feat: #75

* [:lipstick: layout] 챌린지 날짜를 설정할 때 기본 형식과 달이 넘어갈 때의 형식으로 구분해서 보여줌

feat: #75

* [:layout: style] 캘린더 스크립트대로 수정 및 오늘 날짜 디자인 변경

feat: #75

* 알림센터 (#73)

* [:lipstick: layout] 알람센터 셀 생성

feaT: 알림센터 알림유/무 버전 #63

* [:lipstick: layout]  메인 화면 alarm center 진입점 생성

feat: 알림센터 알림유/무 버전 #63

* [:lipstick: layout] 알람센터 셀 생성

feaT: 알림센터 알림유/무 버전 #63

* [:lipstick: layout]  메인 화면 alarm center 진입점 생성

feat: 알림센터 알림유/무 버전 #63

* [:lipstick: layout] alarm center cell 생성

feat: 알림센터 알림유/무 버전 #63

* [:lipstick: layout] 알림 리액션 화면 작업

feat: 알림센터 알림유/무 버전 #63

* [:lipstick: layout] 알람센터 empty view 설정

feat: 알림센터 알림유/무 버전 #63

* [:lipstick: layout] 시작 스토리보드 설정

feat: 알림센터 알림유/무 버전 #63

* [:fire: remove] 네비게이션 버튼 코드 삭제

feat: 알림센터 알림유/무 버전 #63

* [:art: syle] 더미데이터 정리, 알람버튼 액션 정리

feat: 알림센터 알림유/무 버전 #63

* [:lipstick:  layout] 응원, 축하 토스트 화면 구현

feat: 알림센터 알림유/무 버전 #63

* [:fire: remove] whitespace 제거

feat: 알림센터 알림유/무 버전 #63

* [:art: style] setLabel extension 사용

feat: 알림센터 알림유/무 버전 #63

* [:lipstick: layout] 네비 바 와 여백 추가

feat: 알림센터 알림유/무 버전 #63

* [:lipstick: layout] 네비게이션 바 back 버튼 수정

feat: 알림센터 알림유/무 버전 #63

* [:lipstick: layout] 알람 이미지 뷰 설정

feat: 알림센터 알림유/무 버전 #63

* 카카오 로그인 연동 (#76)

* [:lipstick: layout] kakao 로고 이미지 추가

feat: #72

* [:card_file_box: database] lint 제외 문자열 추가

feat: #72

* [:sparkles: feature] 카카오 로그인 시 카카오톡 화면에서 다시 우리 어플로 돌아올 때를 위해 URLContexts 함수 추가

feat: #72

* [:sparkles: feature] Kakao 네이티브 앱 키 추가

feat: #72

* [:lipstick: layout] 카카오 로그인 버튼 생성

* [:sparkles: feature] 카카오 로그인 후 데이터 받아서 서버에 전송

feat: #72

* [:sparkles: feature] HomeVC에도 로그인 연결
body: 테스트 할 때를 위해 주석 처리 해둠

feat: #72

* [:sparkles: feature] 카카오 로그인 및 애플 로그인 기능 추가

* 챌린제 메인 대공사 (#77)

* [:lipstick: layout] 메인 화면 레이아웃 정리

feat: 메인화면 수정 #74

* [:lipstick: layout] EmptyView 설정

feat: 메인화면 수정 #74

* [:sparkles: feature] EmptyView delegate 연결

feat: 메인화면 수정 #74

* [:art: style] Empty view 설정부분 주석 처리

feat: 메인화면 수정 #74

* [:bug: bug] 챌린지 편집 버그 수정

feat: 메인화면 수정 #74

* [:lipstick: layout] empty view 설정

feat: 메인화면 수정 #74

* [:bug: bug] 레이아웃 버그 수정

feat: 메인화면 수정 #74

* [:lipstick: layout] 내 챌린지(미진행) 네트워크 연결

feat: 메인화면 수정 #74

* [:sparkles: feature]  메인화면 네트워크 연결

feat: 메인화면 수정 #74

* [:sparkles: feature] 챌린지 완료 토글 네트워크 연결

feat: 메인화면 수정 #74

* [:sparkles: feature] 내 불편함 변경 APITarget 설정

feat: 메인화면 수정 #74

* [:sparkles: feature] 내 불편함 업데이트 api 설정

feat: 메인화면 수정 #74

* [:art: style] 키보드 세팅 익스텐션 파일 분리

feat: 메인화면 수정 #74

* [:bug: bug] bug fix

feat :메인화면 수정 #74

* [:sparkles: feature] 데이터 불러오기 전 뷰 숨김 처리

feat: 메인화면 수정 #74

* [:sparkles: feature]  다른 유저 챌린지 보기 설정

feat: 메인화면 수정 #74

* [:sparkles:  feature] 온보딩 마지막 단계 넣기

feat: 메인화면 수정 #74

* 회원 탈퇴하기 api 연결 (#80)

* [:sparkles: feature] 유저 삭제(탈퇴) api 연결

feat: #78

* [:art: style] white space 제거

* [:sparkles: feature] 탈퇴 api 연결 후 alert에 연결

feat: #78

* 온보딩/온보딩4 카카오톡 로그인 연결 (#84)

* [:lipstick: layout] 온보딩4 카톡 로그인 디자인

feat: 온보딩/온보딩4 카카오톡 로그인 연결 #79

* [:sparkles: feature] 카카오 로그인 연결하기

feat: 온보딩/온보딩4 카카오톡 로그인 연결 #79

fix #79

* 앱 아이콘 및 어플리케이션 이름 적용 (#83)

* [:lipstick: layout] 앱 아이콘 추가

feat: #81

* [:lipstick: layout] 앱 display name 워시유어보틀 로 수정

feat: #81

* UserDefaultService 만들기 (#85)

* [:sparkles: feature] UserDefaultManager 생성

feat: UserDefaultService 만들기 #82

* [:sparkles: feature] 로그인 토큰 연결

feat: UserDefaultService 만들기 #82

* [:lipstick: layout] 다크모드 제거

* [:bug: bug] 닉세팅 뷰 -> 챌린지 넘어가기 추가

feat" UserDefaultService 만들기 #82

fix: #82

* [: twisted_rightwards_arrows: merge] rebase to develop

* [:twisted_rightwards_arrows: merge] conflict 해결

* [:poop: poop] 서버 테스트코드 작성중

* [:white_check_mark: test] 서버 더미데이터를 대신해서 사용할 json 데이터 추가

feat: #75

* [:sparkles: feature] calendar api에서 id 값을 주지 않으면 기본적으로 현재 챌린지 하고 있는 데이터 전달 받기 위해 optional로 수정

feat: #75

* [:twisted_rightwards_arrows: merge] conflict 해결

* [:bug: bug] Challenge.storyboard conflict 제거

feat: 캘린더 API 연결  #75

* [:lipstick: layout] 블러뷰를 white 뷰로 변경

feat: #75

* [:sparkles: feature] 아래로 내리는 제스쳐 추가

feat: #76

* [:lipstick: layout] 스와이프 바 추가

feat: #75

* [:white_check_mark: test] 캘린더 테스트용 json 수정

feat: #75

* [:art: style] 토큰값을 userdefault에서 불러오는 방식으로 수정

feat: #75

* [:fire: remove] 사용하지 않는 코드 삭제

feat: #75

* [:lipstick: layout] 현재 사용자가 달력을 보는지 팔로워의 달력인지에 따라 challengeView 다르게 구성

feat: #75

* [:art: style] modal에서 self 키워드 삭제

feat: #75

* [:lipstick: layout] 챌린지간 간격 수정

feat: #75

* [:lipstick: layout] 다른 유저의 캘린더에서 현재 진행중인 챌린지가 없을 때 text 변경

feat: #75

* [:lipstick: layout] 버튼 넓이 조절

* [:fire: remove] 합치다가 중복으로 들어간 코드 삭제

feat: #75

* [:sparkles: feature] 캘린더 custom View 로 연결

feat: #75

* [:art: style] guard let을 이용하여 ViewController를 정확히 명시

feat: #75

* [:recycle: refactor] Lint 에러 수정 및 Mark 주석 추가

* [:sparkles: feature] 팔로워의 캘린더를 볼 수 있는 API를 추가 후 분기처리

feat: #75

* [:sparkles: feature] 달력에서 챌린지 참여하기 버튼 클릭 시 기능을 위해 클로저 property 적용

feat: #75

* [:sparkles: feature] 캘린더 VC에서 캘린더 View로 클로저 이동

feat: #75

* [:sparkles: feature] 챌린지 화면에서 캘린더 이동 시 본인/팔로워 구분

feat: #75

* [:art: style] token 이용 시 extension 사용으로 변경

* [:sparkles: feature] 로그아웃 시 토큰 삭제 기능 추가

* [:sparkles: feature] 캘린더 선택 후 데이터를 받아오는 과정에서 챌린지 리스트가 적용되지 않는 부분 해결

feat: #75

Co-authored-by: himygit <43847166+Juhyeoklee@users.noreply.github.com>
Co-authored-by: Juhyeoklee <gur6531@naver.com>

xwoud committed:
[:bug: bug] 합치는 과정에서 생긴 문법 버그 수정

xwoud committed:
[:art: style] userdefault 에 사용하는 token 변수 적용

feat: #60

xwoud committed:
[:sparkles: feature] 보틀월드의 구조체의 challenge를 optional으로 변경한 후 그에 맞게 기능 적용
값이 들어오지 않으면 아직 챌린지를 하고 있지 않다고 판단

feat: #60

xwoud committed:
[:lipstick: layout] 빈 물병 이미지로 변경

feat: #60

xwoud committed:
[:fire: remove] test 코드 삭제

xwoud committed:
[:lipstick: layout] empty view 경우에 맞게 설정

feat: #60

xwoud committed:
[:sparkles: feature] 팔로잉을 거는건지 취소하는 건지 알기 위해 Bool 값을 함께 전달

feat: #60

xwoud committed:
[:sparkles: feature] 팔로잉/언팔로잉 기능의 API 연결 및 서비스 제작

feat: #60

xwoud committed:
[:sparkles: feature] 팔로우, 팔로잉 버튼 클릭 시 네트워크 연결 후 성공 시 데이터 변경

feat: :#60

xwoud committed:
[:sparkles: feature] 새로고침 시 API 다시 불러오기 및 검색어 제거

feat: #60

xwoud committed:
[:sparkles: feature] 보틀월드 데이터를 받아올 때 offset을 설정

feat: #60

xwoud committed:
[:sparkles: feature] 스크롤 시 테이블 뷰 마지막에 닿으면 새롭게 데이터를 불러오는 기능 추가

feat: #60

xwoud committed:
[:art: style] 보틀월드 유저 조회 구조체 추가

feat: #60

xwoud committed:
[:sparkles: feature] 받아오는 데이터 구조체 변경 후 뿌려주기

feat: #60

xwoud committed:
[:sparkles: feature] 팔로우 팔로잉 목록도 동일하게 구조체 변경

feat: #60

xwoud committed:
[:sparkles: feature] 서버 구조체 변경 및 적용

feat: #60

xwoud committed:
[:art: style] 띄어쓰기 및 if문으로의 변경

feat: #60

xwoud committed:
[:lipstick: layout] 네비게이션 바 컬러 변경

feat: #60

xwoud committed:
[:sparkles: feature] 바뀐 api 에서 받은 count로 following, follower 숫자를 변경

feat: #60

xwoud committed:
[:fire: remove] 특정 탭 바로 이동하는 코드 삭제

feat: #60

xwoud committed:
[:art: style] 받아오는 팔로잉 서버 데이터 변수명 변경

feat: #60

xwoud committed:
[:bug: bug] 탭을 이동 시 탭 select 표시가 사라지는 문제 해결

feat: #60

xwoud committed:
[:sparkles: feature] paging 구현 길이 수정

feat: #60

xwoud committed:
[:sparkles: feature] 페이징 시 기존 데이터에 더하는 기능 추가

feat: #60
